src/main/kotlin/no/nav/familie/tilbake/dokumentbestilling/felles/BrevmetadataUtil.kt

```diff
@@ -0,0 +1,142 @@
+                organisasjonService.mapTilInstitusjonForBrevgenerering(it.organisasjonsnummer)
+            }
+        )
+        return brevmetadata
+    }
+
+    fun lagBrevmetadataForMottakerTilForhåndsvisning(
+        behandlingId: UUID,
+    ): Pair<Brevmetadata?, Brevmottager> {
+        val behandling = behandlingRepository.findByIdOrThrow(behandlingId)
+        val fagsak = fagsakRepository.findByIdOrThrow(behandling.fagsakId)
+
+        if (!featureToggleService.isEnabled(FeatureToggleConfig.KONSOLIDERT_HÅNDTERING_AV_BREVMOTTAKERE)) {
+            logger.info("Dropper forhåndsutfylling av metadata som er felles da toggle er avskrudd")
+            return null to BrevmottagerUtil.utledBrevmottager(behandling, fagsak)
+        }
+        val støtterManuelleBrevmottakere: Boolean = BehandlingService.sjekkOmManuelleBrevmottakereErStøttet(
+            behandling = behandling,
+            fagsak = fagsak,
+            featureToggleEnabled = featureToggleService.isEnabled(FeatureToggleConfig.DISTRIBUER_TIL_MANUELLE_BREVMOTTAKERE)
+        )
+        val (bruker, tilleggsmottaker) = DistribusjonshåndteringService.utledMottakere(
+            behandling = behandling,
+            fagsak = fagsak,
+            erManuelleMottakereStøttet = støtterManuelleBrevmottakere,
+            manueltRegistrerteMottakere = manuelleBrevmottakerRepository.findByBehandlingId(behandling.id).toSet()
+        )
+        val (brevmottager, manuellAdresseinfo) = when (tilleggsmottaker) {
+            null -> bruker.somBrevmottager to bruker.manuellAdresse
+            else -> tilleggsmottaker.somBrevmottager to tilleggsmottaker.manuellAdresse
+        }
+        val metadata = genererMetadataForBrev(
+            behandlingId = behandling.id,
+            brevmottager = brevmottager,
+            manuellAdresseinfo = manuellAdresseinfo,
+            annenMottakersNavn = if (tilleggsmottaker != null) {
+                eksterneDataForBrevService.hentPerson(fagsak.bruker.ident, fagsak.fagsystem).navn
+            } else null
+        )
+        return metadata to brevmottager
+    }
+
+    private fun hentAnsvarligSaksbehandlerNavn(
+        persistertSaksbehandlerId: String,
+        vedtaksbrevgrunnlag: Vedtaksbrevgrunnlag?
+    ): String {
+        return if (vedtaksbrevgrunnlag != null) {
+            when (vedtaksbrevgrunnlag.aktivtSteg) {
+                Behandlingssteg.FORESLÅ_VEDTAK ->
+                    eksterneDataForBrevService.hentPåloggetSaksbehandlernavnMedDefault(persistertSaksbehandlerId)
+                else ->
+                    eksterneDataForBrevService.hentSaksbehandlernavn(persistertSaksbehandlerId)
+            }
+        } else if (persistertSaksbehandlerId != Constants.BRUKER_ID_VEDTAKSLØSNINGEN){
+            eksterneDataForBrevService.hentPåloggetSaksbehandlernavnMedDefault(persistertSaksbehandlerId)
+        } else {
+            ""
+        }
+    }
+}
```

Comment:
```suggestion
    private fun hentAnsvarligSaksbehandlerNavn(
        persistertSaksbehandlerId: String,
        vedtaksbrevgrunnlag: Vedtaksbrevgrunnlag?
    ): String {
        return when {
            vedtaksbrevgrunnlag?.aktivtSteg == Behandlingssteg.FORESLÅ_VEDTAK ->
                eksterneDataForBrevService.hentPåloggetSaksbehandlernavnMedDefault(persistertSaksbehandlerId)
            vedtaksbrevgrunnlag != null ->
                eksterneDataForBrevService.hentSaksbehandlernavn(persistertSaksbehandlerId)
            persistertSaksbehandlerId != Constants.BRUKER_ID_VEDTAKSLØSNINGEN ->
                eksterneDataForBrevService.hentPåloggetSaksbehandlernavnMedDefault(persistertSaksbehandlerId)
            else -> ""
        }
    }
```

Usikker om dette er noe bedre, men man kunne ha vurdert å ha alt inne i when.
Så blir det veldig sekvensielt å lese hva som skjer.

Reply:
Liker det bedre 👍